Framework/MDAlgorithms/src/IntegratePeaksMDHKL.cpp

```diff
@@ -247,9 +285,15 @@ void IntegratePeaksMDHKL::integratePeak(const int neighborPts,
       }
     }
   }
-  double ratio = float(peakPoints) / float(measuredPoints - peakPoints);
-  intensity = peakSum - ratio * (measuredSum - peakSum);
-  errorSquared = errSqSum + ratio * ratio * (measuredErrSqSum - errSqSum);
+  if (BackgroundOuterRadius2 > 0.0) {
+    double ratio = float(peakPoints) / float(backgroundPoints);
```

Comment:
If I set the values so zero `backgroundPoints` are selected then here it's dividing by zero and the resulting intensity is `-nan`.